### PR TITLE
[Docs] Creates Developer docs GH page code

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description

Uses Docsify to generate documentation liks that expose Readme's within the OSD repo as HTML.

The GH Pages running on my local fork: https://ashwin-pc.github.io/OpenSearch-Dashboards/docs/index.html#/

## Screenshot

Screenshot 2024-02-27 at 6 17 02 PM

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/discover/public/url_generator.ts [hunk 1]
```diff
@@ -122,6 +122,7 @@ export class DiscoverUrlGenerator
     url = setStateToOsdUrl<QueryState>('_g', queryState, { useHash }, url);
     url = setStateToOsdUrl('_a', appState, { useHash }, url);
 
+    // debugger;
```
Comment (SuZhou-Joe) [inline src/plugins/discover/public/url_generator.ts:125]: This line should be removed I suppose.

File: docs/_sidebar.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    - [Opensearch dashboards.release notes 1.3.8](../release-notes/opensearch-dashboards.release-notes-1.3.8.md)
+    - [Opensearch dashboards.release notes 1.3.9](../release-notes/opensearch-dashboards.release-notes-1.3.9.md)
+    - [Opensearch dashboards.release notes 2.0.0 rc1](../release-notes/opensearch-dashboards.release-notes-2.0.0-rc1.md)
+    - [Opensearch dashboards.release notes 2.0.0](../release-notes/opensearch-dashboards.release-notes-2.0.0.md)
+    - [Opensearch dashboards.release notes 2.0.1](../release-notes/opensearch-dashboards.release-notes-2.0.1.md)
+    - [Opensearch dashboards.release notes 2.1.0](../release-notes/opensearch-dashboards.release-notes-2.1.0.md)
+    - [Opensearch dashboards.release notes 2.10.0](../release-notes/opensearch-dashboards.release-notes-2.10.0.md)
+    - [Opensearch dashboards.release notes 2.11.0](../release-notes/opensearch-dashboards.release-notes-2.11.0.md)
+    - [Opensearch dashboards.release notes 2.11.1](../release-notes/opensearch-dashboards.release-notes-2.11.1.md)
+    - [Opensearch dashboards.release notes 2.12.0](../release-notes/opensearch-dashboards.release-notes-2.12.0.md)
```
Comment (SuZhou-Joe) [inline docs/_sidebar.md:153]: A little bit weird about the order here.

File: docs/_sidebar.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    - [Opensearch dashboards.release notes 1.0.0 rc1](../release-notes/opensearch-dashboards.release-notes-1.0.0-rc1.md)
+    - [Opensearch dashboards.release notes 1.0.0](../release-notes/opensearch-dashboards.release-notes-1.0.0.md)
+    - [Opensearch dashboards.release notes 1.1.0](../release-notes/opensearch-dashboards.release-notes-1.1.0.md)
+    - [Opensearch dashboards.release notes 1.2.0](../release-notes/opensearch-dashboards.release-notes-1.2.0.md)
+    - [Opensearch dashboards.release notes 1.3.0](../release-notes/opensearch-dashboards.release-notes-1.3.0.md)
+    - [Opensearch dashboards.release notes 1.3.1](../release-notes/opensearch-dashboards.release-notes-1.3.1.md)
+    - [Opensearch dashboards.release notes 1.3.10](../release-notes/opensearch-dashboards.release-notes-1.3.10.md)
+    - [Opensearch dashboards.release notes 1.3.11](../release-notes/opensearch-dashboards.release-notes-1.3.11.md)
+    - [Opensearch dashboards.release notes 1.3.12](../release-notes/opensearch-dashboards.release-notes-1.3.12.md)
+    - [Opensearch dashboards.release notes 1.3.13](../release-notes/opensearch-dashboards.release-notes-1.3.13.md)
```
Comment (SuZhou-Joe) [inline docs/_sidebar.md:137]: Nit: order need to adjust I suppose

File: docs/_sidebar.md (hunk 2 above)
Comment (ashwin-pc) [reply docs/_sidebar.md:153]: Dont quite know what you mean by the order here. This is simply the folder structure of the OSD repository. What did you expect here?

File: src/plugins/discover/public/url_generator.ts (hunk 1 above)
Comment (ashwin-pc) [reply src/plugins/discover/public/url_generator.ts:125]: Good catch, thanks!

**Review by SuZhou-Joe — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Comment by kavilla** [comment]
good stuff!

**Pull request merged**